### ISSUE TITLE
Tab in string of json causes MQTT node failure when output is JSON object

### ISSUE THREAD

**Issue by jeroenst** [node-red/node-red#3071]
### Current Behavior

![image](https://user-images.githubusercontent.com/2833940/126027677-31a8ba9b-3840-476b-aca4-28631e29664a.png)

### Expected Behavior

No error when a tab is in the string of a json object

### Steps To Reproduce

Send this string to an mqtt-in node which outputs a JSON object (notice there is a tab after the minimal.bin.gz):

{"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota-minimal.bin.gz	","RestartReason":"Software/System restart","Uptime":"0T21:03:06","StartupUTC":"2021-07-16T09:05:29","Sleep":50,"CfgHolder":4617,"BootCount":17,"BCResetTime":"2021-06-12T10:53:10","SaveCount":23,"SaveAddress":"F7000"}}

### Example flow
[mqttjsontaberrorflow.json.zip](https://github.com/node-red/node-red/files/6834275/mqttjsontaberrorflow.json.zip)
```
[{"id":"68704a13.2cc684","type":"mqtt in","z":"bfd0a828.fc43d8","name":"","topic":"home/test","qos":"1","datatype":"json","broker":"95a2d9cb.10d398","nl":false,"rap":false,"x":460,"y":4220,"wires":[["668bf7d8.b49078"]]},{"id":"e2f6768d.2e4e88","type":"mqtt out","z":"bfd0a828.fc43d8","name":"","topic":"","qos":"1","retain":"true","respTopic":"","contentType":"","userProps":"","correl":"","expiry":"","broker":"95a2d9cb.10d398","x":270,"y":4220,"wires":[]},{"id":"668bf7d8.b49078","type":"debug","z":"bfd0a828.fc43d8","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"false","statusVal":"","statusType":"auto","x":650,"y":4220,"wires":[]},{"id":"27ce3f32.17493","type":"inject","z":"bfd0a828.fc43d8","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"home/test","payload":"{\"StatusPRM\":{\"Baudrate\":115200,\"SerialConfig\":\"8N1\",\"GroupTopic\":\"tasmotas\",\"OtaUrl\":\"http://ota.tasmota.com/tasmota/release/tasmota-minimal.bin.gz\t\",\"RestartReason\":\"Software/System restart\",\"Uptime\":\"0T21:09:18\",\"StartupUTC\":\"2021-07-16T09:05:29\",\"Sleep\":50,\"CfgHolder\":4617,\"BootCount\":17,\"BCResetTime\":\"2021-06-12T10:53:10\",\"SaveCount\":23,\"SaveAddress\":\"F7000\"}}","payloadType":"str","x":120,"y":4220,"wires":[["e2f6768d.2e4e88"]]},{"id":"95a2d9cb.10d398","type":"mqtt-broker","name":"Server","broker":"jeroensteenhuis.nl","port":"1883","clientid":"","usetls":false,"compatmode":false,"protocolVersion":4,"keepalive":"60","cleansession":true,"birthTopic":"","birthQos":"0","birthPayload":"","closeTopic":"","closeQos":"0","closePayload":"","willTopic":"","willQos":"0","willPayload":""}]
```

### Environment

- Node-RED version: 1.3.5
- Node.js version: 10.24.1
- npm version: 6.14.12
- Platform/OS: Linux Mint 20.2
- Browser: Chrome

**Comment by hardillb** [comment]
Raw tabs in strings are not valid JSON, they should be encoded as `\t`

**Comment by jeroenst** [comment]
Then it is an error in https://github.com/arendst/Tasmota which sends this string to nodered.

**Comment by jeroenst** [comment]
> Raw tabs in string values are not valid JSON, they should be encoded as `\t`, so this is the correct behaviour for the JSON decoder

There is a tab in the string according to the flow:

"http://ota.tasmota.com/tasmota/release/tasmota-minimal.bin.gz\t\"

Or is this incorrect?

**Comment by hardillb** [comment]
Probably, a tab at the end of a URL doesn't make any sense either, so I can't see a good reason for them to send it

**Comment by jeroenst** [comment]
True, I don't know why the tab is there, but after all I think tab's in json strings shouldn't lead to a parsing error.

**Comment by hardillb** [comment]
It is a parsing error, because the input is not valid JSON (also the parsing is being done by NodeJS not Node-RED since it's just calling `JSON.parse()`)

**Comment by hardillb** [comment]
Create a text file with your input json (with the tab) called `tab.json` and run the following small node app:

```javascript
var fs = require('fs')
fs.readFile('tab.json', function(err, data) {
  console.log(JSON.parse(data))
})
```

**Comment by knolleary** [comment]
You might need to double escape the tab - it might be getting converted to a non escape tab somewhere in the path from the editor input box all the way to the flow config.

Try `\\t`

**Comment by jeroenst** [comment]
I get this error using the node code:

internal/modules/cjs/loader.js:807
  return process.dlopen(module, path.toNamespacedPath(filename));
                 ^

Error: /root/tab.node: invalid ELF header
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:807:18)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)

**Comment by hardillb** [comment]
No idea how you go that error, the code I posted should be in a file called `test.js` and run with `node test.js` in the same directory as the `tab.json` file.

Anyway, as previously mentioned the MQTT input message is invalid JSON so the parsing error is correct, and you need to ask the Tasmota project why they are sending the tab at the end of the URL.

As a workaround you can probably tell the MQTT-in node to output a string and then use a function node to remove the tab character before passing the output through a JSON node to parse the string.

**Comment by hardillb** [comment]
Example function node:

```javascript
msg.payload = msg.payload.replace("\t","")
return msg;
```

**Comment by jeroenst** [comment]
Got it, file extension I used was .node and had to be .js.

Output is:

undefined:1
{"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota-minimal.bin.gz	","RestartReason":"Software/System restart","Uptime":"0T21:09:18","StartupUTC":"2021-07-16T09:05:29","Sleep":50,"CfgHolder":4617,"BootCount":17,"BCResetTime":"2021-06-12T10:53:10","SaveCount":23,"SaveAddress":"F7000"}}
                                                                                                                                                    ^

SyntaxError: Unexpected token 	 in JSON at position 148
    at JSON.parse (<anonymous>)
    at /root/tab.js:3:20
    at FSReqWrap.readFileAfterClose [as oncomplete] (internal/fs/read_file_context.js:53:3)

So indeed the tab character is not supported in the json parser of nodejs.

Also I can confirm it is a tab character instead of an encoded tab (\t) in the string received from https://github.com/arendst/Tasmota

I have opened a question in Tasmota for this problem to see how it can be solved.

Thank you for your support!
